index.js

```diff
@@ -96,101 +157,66 @@ module.exports = function(gulp) {
-        log.warn('Message: ' + err.message);
-      }
-    }))
-    .pipe(gulp.dest(paths.dist))
-  };
-  gulp.task('build-styles', buildStyles);
-
-  /* Reduce all javascript to app.js */
-  buildApp = function() {
-    return browserifyBundler.bundle()
-      .on('error', function(err, b) {
-        delete err.stream;
-        log.error('[BROWSERIFY] @ ' + currentDateTime());
-        log.warn(err.toString());
-        return true;
-      })
-      .pipe(source('app.js'))
-      .pipe(buffer())
-      .pipe($.util.env.type !== 'production' ? $.sourcemaps.init({loadMaps: true}) : $.util.noop())
-      .pipe($.util.env.type === 'production' ? $.uglify() : $.util.noop())
-      .pipe($.util.env.type !== 'production' ? $.sourcemaps.write('./') : $.util.noop())
-      .pipe(gulp.dest(paths.dist + '/' + paths.scripts));
-  };
-  gulp.task('build-app', buildApp);
-
-  /* Full build */
-  gulp.task('build', function(callback) {
-    browserifyBundler = browserify(watchify.args);
-    configureBrowserify();
-    runSequence('clean',
-      [
-        'copy-resources',
-        'build-styles',
-        'build-app'
-      ], function() {
-        log.mark('[BROWSERIFY] complete!');
-        callback();
-      });
-  });
-
   /* Watch build */
   gulp.task('watch', function() {
-    browserifyBundler = watchify(browserify(watchify.args));
-    configureBrowserify();
-    gulp.watch(paths.app + '/' + paths.styles + '/**/*.scss', buildStyles);
-    gulp.watch([
-      paths.app + '/**/*.*',
-      '!' + paths.app + '/**/*.+(' + nonResources +')',
-    ], copyResource);
-    browserifyBundler.on('update', buildApp);
-    browserifyBundler.on('log', function(data) {
-      log.mark('[BROWSERIFY] ' + data.toString());
-    });
-    runSequence('clean',
-      [
-        'copy-resources',
-        'build-styles',
-        'build-app'
-      ]);
+    if (!options.styles.skip) {
+      gulp.watch(options.styles.src, ['build-styles']);
+    }
+
+    if (!options.copy.skip) {
+      gulp.watch([
+        options.copy.src + '/**/*.*',
+        '!' + options.copy.src + '/**/*.+(' + options.copy.excludes +')',
+      ], copyResource);
+    }
+
+    if (!options.browserify.skip) {
+      browserifyTask.createWatchifyBundler();
+    }
+
+    if (options.clean.skip) {
+      runSequence(buildTasks);
+    } else {
+      runSequence(options.taskPrefix + 'clean', buildTasks);
+    }
   });
 
   /* Returns public configuration API */
   return {
-    transform: function(fn) {
-      transformStack.push(fn);
-    },
-
-    nonResources: function(nrsc) {
-      nonResources = nrsc;
+    /**
+     * @method transform
+     * @param transform {String or Object} either a string referencing the transform to perform or an object
+     *                  can either be a string identifying the transform or an object of the format:
+     *                    {name: 'name of transform', options: {// transform options}}
+     */
+    transform: function(transform) {
```

Comment:
now that you've added ability to set options in the constructor, is it necessary to have any of these public API calls?